File: passenger-service/src/main/java/com/internship/passengerservice/utils/exceptions/ExceptionCodes.java
package com.internship.passengerservice.utils.exceptions;

import lombok.AllArgsConstructor;
import lombok.Getter;

@AllArgsConstructor
@Getter
public enum ExceptionCodes {
    PASSENGER_NOT_FOUND("passenger.notFound"),
    PASSENGER_ID_IS_NULL("passenger.id.isNull"),
    RATE_NOT_FOUND("rate.notFound"),
    ERROR_INVALID_INPUT("error.input.invalid"),
    ERROR_NOT_READABLE("error.notReadable"),
    PHONE_ALREADY_EXISTS("phone.alreadyExists"),
    EMAIL_ALREADY_EXISTS("email.alreadyExists"),
    TYPE_MISMATCH("type.mismatch"),
    UNKNOWN_ERROR("error.unknown");
    private final String code;
}

File: finance-service/src/main/kotlin/com/internship/financeservice/utils/exceptions/ExceptionCodes.kt
package com.internship.financeservice.utils.exceptions

enum class ExceptionCodes(private val code: String) {
    ERROR_NOT_READABLE("error.notReadable"),
    ERROR_INVALID_INPUT("error.invalidInput"),
    WALLET_NOT_FOUND("wallet.notFound"),
    CARD_NOT_FOUND("card.notFound"),
    PAYMENT_NOT_FOUND("payment.notFound"),
    WALLET_TRANSFER_NOT_FOUND("walletTransfer.notFound"),
    WALLET_TRANSFER_INVALID_AMOUNT("wallet.transfer.invalidAmount");

    fun getCode(): String {
        return code
    }
}